### PR TITLE
fix(ux): Hide Zowe Resources panel until content is shown

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Hides the "Zowe Resources" panel until a table view is provided to the `TableViewProvider`.
We (as well as extenders) can also hide the panel by invoking `TableViewProvider.setTableView` with `null` as the parameter.

## Release Notes

Milestone: vNext

Changelog:

- Added placeholder text to the "Zowe Resources" panel to provide clarity when a table is not visible.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by t1m0thyj — DISMISSED** [review]
LGTM, thanks @traeok!

**Review by zFernand0 — DISMISSED** [review]
I like this! 🥳 

I think this could be very useful for the z/OS console view. 😅 

LGTM! 😋

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 ### Bug fixes
 
+- The "Zowe Resources" panel is now hidden by default until it is used to display a table or other data. [#3113](https://github.com/zowe/zowe-explorer-vscode/issues/3113)
```
Comment (anaxceron) [inline packages/zowe-explorer/CHANGELOG.md:11]: Can we reword this to say: " ... until the user opens the panel to display a table or other data." I'm trying to use active voice here.

Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md:11]: Zowe Explorer will open the panel automatically when displaying a table or other data. That said, I will update the changelog to clarify this.

Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md]: Addressed in 5830d76

**Review by awharn — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]

**Review by JillieBeanSim — APPROVED** [review]
Thanks so much @traeok this looks great!
